web/modules/custom/joinup_core/src/JoinupRelationManager.php

```diff
@@ -219,25 +227,60 @@ public function getSolutionIds(): array {
    *   An array of entity IDs.
    */
   protected function getRdfEntityIdsByBundle(string $bundle): array {
-    try {
-      // Since the Joinup Core module depends on the RDF Entity module we can
-      // reasonably assume that the entity storage is defined and is valid. If
-      // it is not this is due to exceptional circumstances occuring at runtime.
-      $storage = $this->entityTypeManager->getStorage('rdf_entity');
-      $definition = $this->entityTypeManager->getDefinition('rdf_entity');
-    }
-    catch (InvalidPluginDefinitionException $e) {
-      throw new \RuntimeException('The RDF entity storage is not valid.');
-    }
-    catch (PluginNotFoundException $e) {
-      throw new \RuntimeException('The RDF entity storage is not defined.');
-    }
-
+    $storage = $this->getRdfStorage();
+    $definition = $this->entityTypeManager->getDefinition('rdf_entity');
     $bundle_key = $definition->getKey('bundle');
 
     $query = $storage->getQuery();
     $query->condition($bundle_key, $bundle);
     return $query->execute();
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function getContactInformationRelatedGroups(RdfInterface $entity): array {
+    // When the user creates a group, they do not have any roles in the group
+    // yet. There is no need to have a check for groups when the entity is new.
+    if ($entity->isNew()) {
+      return [];
+    }
+
+    $query = $this->getRdfStorage()->getQuery();
+    $condition_or = $query->orConditionGroup();
+    // Contact entities are also referenced by releases but this value is
+    // inherited by the solution directly so there is no need to check them.
+    $condition_or->condition('field_ar_contact_information', $entity->id());
+    $condition_or->condition('field_is_contact_information', $entity->id());
+    $query->condition($condition_or);
+    $ids = $query->execute();
+
+    return empty($ids) ? [] : $this->getRdfStorage()->loadMultiple($ids);
+  }
+
+  /**
+   * Returns the rdf storage class.
+   *
+   * @return \Drupal\sparql_entity_storage\SparqlEntityStorageInterface
+   *   The rdf storage class.
+   */
+  protected function getRdfStorage(): SparqlEntityStorageInterface {
+    if (empty($this->sparqlStorage)) {
+      try {
+        // Since the Joinup Core module depends on the RDF Entity module we can
+        // reasonably assume that the entity storage is defined and is valid. If
+        // it is not this is due to exceptional circumstances occurring at
+        // runtime.
+        $this->sparqlStorage = $this->entityTypeManager->getStorage('rdf_entity');
+      }
+      catch (InvalidPluginDefinitionException $e) {
+        throw new \RuntimeException('The RDF entity storage is not valid.');
+      }
+      catch (PluginNotFoundException $e) {
+        throw new \RuntimeException('The RDF entity storage is not defined.');
+      }
+    }
+    return $this->sparqlStorage;
+  }
```

Comment:
Don't provide a static cache for the storage, it is already statically cached inside the `EntityTypeManager`. Having this here is a potential source of very difficult to track caching issues.

This not only increases our memory footprint without being necessary, but this not be updated when `EntityTypeManager::clearCachedDefinitions()` is called and will return stale results. Also this does not respect `EntityTypeManager::useCaches(FALSE)` which is called during dangerous operations like update hooks and entity storage schema changes.